File: Scripts/DelayedSpawn.cs

public class DelayedSpawn : MonoBehaviour {
    float delay = 0.5f; // save
	GameObject[] objectsToSpawn;
	bool despawnInstead = false;
	bool doSelfAfterList = false;
	bool destroyAfterListInsteadOfDeactivate = false;
	float timerFinished;
	bool active;

	void OnEnable() {
		if (PauseScript.a != null) timerFinished = Sys_Global.pauseRelativeTime + delay;
        else timerFinished = delay;

		active = true;
    }

	void DelayedSpawnUpdate() {
		if (!active) return;
		if (timerFinished >= Sys_Global.pauseRelativeTime) return;

		active = false; // Once only, unless we do self after the list.
		for (int i=0;i<objectsToSpawn.Length;i++) {
			if (despawnInstead) {
				if (objectsToSpawn[i] != null) flag_set(&instances[objectsToSpawn[i]].entflags, ENTFLAG_ACTIVE, false);
			} else {
				if (objectsToSpawn[i] != null) flag_set(&instances[objectsToSpawn[i]].entflags, ENTFLAG_ACTIVE, true);
			}
		}

		if (doSelfAfterList) {
			if (despawnInstead) {
				if (destroyAfterListInsteadOfDeactivate) {
					DeleteInstance(selfIdx);
				} else {
                    flag_set(&SELF.entflags, ENTFLAG_ACTIVE, false);
					flag_set(&SELF.entflags, ENTFLAG_ACTIVE, false);
				}
			} else {
				gameObject.SetActive(true);
			}
		}
    }
}

File: Scripts/delayedspawn.c
#include "mod.h"

void DelayedSpawnEnable(uint16_t self) {
    Entity* e = &Eng_Global->instances[self];
    e->timerFinished = Eng_Global->pauseRelativeTime + e->delay;
    e->active = true;
}

void DelayedSpawnUpdate(uint16_t self) {
    Entity* e = &Eng_Global->instances[self];
    if (!e->active || e->timerFinished >= Eng_Global->pauseRelativeTime) return;
    e->active = false;
    for (uint8_t i = 0; i < MAX_CHILD_COUNT; i++) {
        uint16_t child = e->child[i];
        if (child == UINT16_MAX) continue; // TODO child[] is the DelayedSpawn object list replacement in the C port.
        flag_set(&Eng_Global->instances[child].entflags,ENTFLAG_ACTIVE,!e->despawnInstead);
    }
    if (!e->doSelfAfterList) return;
    if (e->despawnInstead) {
        if (e->destroyAfterListInsteadOfDeactivate) DeleteInstance(self);
        else flag_set(&e->entflags,ENTFLAG_ACTIVE,false);
    } else flag_set(&e->entflags,ENTFLAG_ACTIVE,true);
}
